### ISSUE TITLE
[Feature]: Async Exp(1) noise pool — precompute the Gumbel noise off the hot path

### ISSUE THREAD

**Issue by dmholtz** [torch-spyre/spyre-inference#919]
### Feature description

Sampling random numbers with `torch.exponential_` is independent of a request (Asterix: assuming no RNG seed is set by the user). For unseeded requests, these random numbers can by asynchronously computed on CPU while the transformer model runs on spyre. As the control flow reaches the sampler, random numbers can be accessed efficiently via a zero-copy view into the precomputed buffer, reducing the `expontial_` number access cost from milliseconds to microseconds.

### Motivation and context

Today the sampler runs entirely on CPU: the model produces logits on Spyre, they are copied to the host, and all sampling (temperature, top-k/top-p, penalties, greedy/random selection, logprobs) runs on CPU.
Drawing random numbers with `torch.exponential_` on CPU is inefficient and bottlenecks the hot decode path, leading to increased ITL. Further, this effort (badly) scales with batch and vocabulary size.

Part of Epic #903 

### Proposed solution

See sendnn-stack implementation: https://github.com/torch-spyre/sendnn-inference/pull/1046

### Checklist

- [x] I have searched for similar feature requests
